From: Krzysztof Wojcik <krzysztof.wojcik@intel.com>
Date: Wed, 20 Apr 2011 15:39:53 +1000
Subject: [PATCH] md: Cleanup after raid45->raid0 takeover
Git-commit: fee68723cf6ae00082f70f3eff17fceab2a4f7d7
Patch-mainline: v2.6.39-rc5

Problem:
After raid4->raid0 takeover operation, another takeover operation
(e.g raid0->raid10) results "kernel oops".
Root cause:
Variables 'degraded' in mddev structure is not cleared
on raid45->raid0 takeover.

This patch reset this variable.

Signed-off-by: Krzysztof Wojcik <krzysztof.wojcik@intel.com>
Signed-off-by: NeilBrown <neilb@suse.de>
---
 drivers/md/md.c |    1 +
 1 file changed, 1 insertion(+)

--- linux-2.6.32-SLE11-SP1.orig/drivers/md/md.c
+++ linux-2.6.32-SLE11-SP1/drivers/md/md.c
@@ -3146,6 +3146,7 @@ level_store(mddev_t *mddev, const char *
 	mddev->layout = mddev->new_layout;
 	mddev->chunk_sectors = mddev->new_chunk_sectors;
 	mddev->delta_disks = 0;
+	mddev->degraded = 0;
 	if (mddev->pers->sync_request == NULL) {
 		/* this is now an array without redundancy, so
 		 * it must always be in_sync
